src/query/atomWithQuery.ts

```diff
@@ -27,7 +28,8 @@ export function atomWithQuery<
     | ((
         get: Getter
       ) => AtomWithQueryOptions<TQueryFnData, TError, TData, TQueryData>),
-  equalityFn: (a: TData, b: TData) => boolean = Object.is
+  equalityFn: (a: TData, b: TData) => boolean = Object.is,
+  getQueryClient: (get: Getter) => QueryClient = (get) => get(queryClientAtom)
```

Comment:
I know there is a preference to avoid objects being passed, but I think this will result in a lot of `atomWithQuery(createQuery, null, (get) => get(myQueryClientAtom))`, as I think passing a custom query client will be used more often than an equalityFn. and maybe passing null is ok, too :)

Reply:
I see your point and yes to the preference.
I'm not sure how often people use `myQueryClientAtom` in app code. Unlike `jotai/urql` which require "URL", the default client in `jotai/query` would work for most cases?
We still can use `<Provider initialValues={[[queryClientAtom, myQueryClient]]}>`.

To allow `null` for equalityFn, we can't use default parameter, right? we can change that.

Reply:
> We still can use <Provider initialValues={[[queryClientAtom, myQueryClient]]}>.

If we did this though, it would not allow you to modify the scope, right?

I think folks might use the custom query client often, esp if they have an app that is using react-query already as they might have defined defaults or other settings they want to use.

Reply:
Yes, for the custom atom scope, we need to use the third argument.
Do you think it's a good idea to swap the second and third arguments? (=breaking change)

Reply:
yeah, I do. I'm honestly not sure how much use or importance the equalityFn provides -- i find modifying the query client to only update on certain changes is more effective https://react-query.tanstack.com/guides/migrating-to-react-query-3#the-queryoptionsnotifyonstatuschange-option-has-been-superceded-by-the-new-notifyonchangeprops-and-notifyonchangepropsexclusions-options

Reply:
If we can remove `equalityFn` completely, it seems better to me. Is it possible?

Reply:
#592